src/workloads/query/PercentilesSpillToDisk.yml

```diff
@@ -0,0 +1,158 @@
+  window function.
+  3. Both the $sort and $_internalSetWindowsFields spill.
+  There is a workload in SetWindowFieldsUnbounded.yml that handles the case if only the $sort spills,
+  and since there is no unique behavior for $percentile in this case, it was not tested in this file.
+  We can use an index to control whether or not the $sort will spill, and we change the size of the
+  partition to control whether or not the $_internalSetWindowsFields will spill.
+
+Keywords:
+- setWindowFields
+- percentile
+- spill
+
+AutoRun:
+- When:
+    mongodb_setup:
+      $eq:
+      - standalone-all-feature-flags
+    branch_name:
+      $neq:
+      - v4.0
+      - v4.2
+      - v4.4
+      - v5.0
+      - v6.0
+
+GlobalDefaults:
+  MaxPhases: &maxPhases 8
+  Database: &db percentiles
+  Collection: &coll Collection0
+
+ActorTemplates:
+- TemplateName: ComputePercentile
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "ComputePercentile"}}
+    Type: CrudActor
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: 1024}}]
+        NopInPhasesUpTo: *maxPhases
+        PhaseConfig:
+          Repeat: 10
+          Database: *db
+          Collection: *coll
+          Operations:
+          - OperationName: aggregate
+            OperationCommand:
+              Pipeline: [
+                {
+                  $setWindowFields:
+                    {
+                      partitionBy: {^Parameter: {Name: "Partition", Default: null}},
+                      sortBy: {^Parameter: {Name: "Sort", Default: {not_spill_indexed: 1}}},
+                      output: {
+                        p: {
+                          $percentile: {p: [0.75], input: "$data", method: "approximate" },
+                          window: { documents: [0, "unbounded"] }
+                        }
+                      }
+                    }
+                }
+              ]
+              Options:
+                AllowDiskUse: {^Parameter: {Name: "Disk", Default: true}}
+
+Actors:
+- Name: CreateDataset
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *maxPhases
+      PhaseConfig:
+        Threads: 1
+        Repeat: 1
+        Database: *db
+        CollectionCount: 1
+        # We have 100k documents with each being between 10kB to 20kB. When divided into 20 partitions, it won't spill, but when divided into fewer partitions (like 2) it will spill.
+        DocumentCount: 100000
+        BatchSize: 1000
+        Document:
+          # We will use the _id field to create evenly distributed partition keys to test spilling.
+          _id: {^Inc: {start: 0}}
+          # This data field is what we calculate the percentiles of.
+          data: {^RandomDouble: {distribution: normal, mean: 50.0, sigma: 10.0}}
+          # This string pads the doc size out to around 10KB.
+          string: &string {^RandomString: {length: 10000}}
```

Comment:
I don't understand the mechanics here: the "string" field isn't used in $setWindowFields. Why does it affect spilling?

Reply:
When we partition the documents, entire documents are spilled to disk. So even though we don't use the `string` field in the `$setWindowFields` stage, the stage will add the entire document to a cache and then spill to disk if it runs out of memory. So the stage will still spill to disk.

I can change this to have more documents without the `string` field. I would just need to add around 1.25 million (I think) too have enough `double` values to spill.

Reply:
The confusing bit for me was that I expected the $percentile window function to trigger spilling (because it accumulates values) but if I understand you correctly, the spilling might happen even for a window function that doesn't collect any data because the overall window functions infra causes the whole document to be kept in memory, and, therefore, might cause spilling. Right?

Reply:
Yes I think so, the pulling in documents happens at a higher level (on the `setWindowFields` stage), so yes I think spilling could happen even if the window function doesn't do anything with data.